src/spatialdata/models/models.py

```diff
@@ -678,6 +732,27 @@ def validate(
         if len(set(expected_regions).symmetric_difference(set(found_regions))) > 0:
```

Comment:
This can be slow, we can make the user aware that this column should be categorical by raising a warning here. If it's categorical we can check the levels, if it's not we can check the uniqueness.

Reply:
Actually the levels is not enought, we should check the counts because there could be a level without a value. Anyway I think with categorical is slightly faster. Or maybe not, wdyt?

Reply:
I mean how many regions we expect? maybe order of thousands (although currently not more than dozens)? I would maybe change the line above to 
`data.obs[attr[self.REGION_KEY_KEY]].cat.categories` which would raise error directly if not categorical

Reply:
Not many regions, but the uniqueness is computed on the rows of the table, which can be a large number.

> data.obs[attr[self.REGION_KEY_KEY]].cat.categories which would raise error directly if not categorical

This could work.

But actually I realized that the most low effort change to apply here is to simply change the error into a warning, and the eventually follow up in a different PR.

Reply:
Currently we do raise the warning that the column is changed to categorical so I implemented it with this current behaviour in mind that so far we did not expect a user to make the column categorical until parsed.

I am not certain here whether a warning would be sufficient given that any downstream functionality depending on this would have to check again and raise an error if not symmetric.

My question, should we perhaps enforce the column to be categorical at this stage already?

Reply:
> My question, should we perhaps enforce the column to be categorical at this stage already?

Yes I think that we can enforce the categorical column already, it would simplify things.

> I am not certain here whether a warning would be sufficient given that any downstream functionality depending on this would have to check again and raise an error if not symmetric.

My opinion is that downstream methods should not rely on this design because it's unnecessarily complex and because methods should be flexible anyway. For instance imagine that a table is annotating `sample0` and `sample1`, and the user asks for the plot of `sample1`. Probably internally the table is subset, so it doesn't really matter if the table had `sample0`, `sample1` or even a `sample2` in `region`, because the subset, and the matching of the rows with the geometries (another filtering + reordering) is required anyway. So the information on `region` is not useful. And if the information is really needed one can still check the values of the categorical column. Not all the categories are guaranteed to have rows with that values, but again if this is really really useful, the user can always call `unique()` manually.

Reply:
I consider the effort to keep those values always in sync unnecessary because most of the time that value will not be used. So better to allow for it to be out-of-sync (and raise a warning), or even better to drop it (follow up PR), and let the user compute it when it's really needed.

Reply:
Internally the table would be subset, but then would we even validate the table at that stage?

Reply:
As discussed here: https://github.com/scverse/spatialdata/pull/410#discussion_r1443007852 we are addressing this in a follow up PR https://github.com/scverse/spatialdata/issues/419 and allowing a gentler handling of the mismatch here (warning instead of error).